service/history/execution/context_test.go

```diff
@@ -3101,3 +3112,75 @@ func TestUpdateWorkflowExecutionWithNewAsPassive(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateWorkflowRequestsAndMode(t *testing.T) {
```

Comment:
Will you have a unit test that covers the error case in the conflict resolution flow for this change when that's updated?